src/Hosting/Hosting/src/Internal/HostingApplicationDiagnostics.cs

```diff
@@ -46,14 +50,24 @@ public void BeginRequest(HttpContext httpContext, HostingApplication.Context con
             }
 
             var diagnosticListenerEnabled = _diagnosticListener.IsEnabled();
+            var diagnosticListenerActivityCreationEnabled = (diagnosticListenerEnabled && _diagnosticListener.IsEnabled(ActivityName, httpContext));
             var loggingEnabled = _logger.IsEnabled(LogLevel.Critical);
 
-            if (loggingEnabled || (diagnosticListenerEnabled && _diagnosticListener.IsEnabled(ActivityName, httpContext)))
+            // TODO: If loggingEnabled check to change to if ActivityPropogation options are enabled
+            if (loggingEnabled || diagnosticListenerActivityCreationEnabled)
             {
-                context.Activity = StartActivity(httpContext, out var hasDiagnosticListener);
-                context.HasDiagnosticListener = hasDiagnosticListener;
+                _dummyListener = new ActivityListener()
```

Comment:
One of the issue is that Activities created using `new Activity()` don't have `ActivityKind` and `ActivitySource` correctly populated.

Reply:
The only reason we are falling back to new Activity() is to support back-compat for pre-existing scenarios, and ActivityKind + ActivitySource aren't part of the pre-existing scenario. I would propose that anyone expecting to see ActivityKind + ActivitySource populated should be using ActivityListener. Does that work or it still causes problems?

Reply:
The diagnostic listener subscriber will see different behavior based on whether there is another listener in the process.
I'm not a fan of having unrelated subscribers cause side-effects

Reply:
would it matter for the diagnostic listeners? we encourage using ActivityListener moving forward. So diagnostic listeners shouldn't care about the Kind anyway. right? or you mean something else?

---
In reply to: [576413791](https://github.com/dotnet/aspnetcore/pull/30089#discussion_r576413791) [](ancestors = 576413791)

Reply:
Perhaps? Not sure.

OpenTelemetry for one, uses a diagnostic listener and then uses private reflection to set the activity kind and activity source on the activity it receives- https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry/Trace/ActivitySourceAdapter.cs

Reply:
The change here is to allow the OTel to use ActivityListener instead of Diagnostics listener. If they continue using Diagnostic Listener, they can continue do what they are doing today but I'll question at that time what is the point of this change we are doing here. I understand they may need to use the Diagnostic Listener only if they decide to support old versions of aspnet core. but this would be a different issue that will not affect the decision here. 

CC @cijothomas

Reply:
Can we add this to our weekly sync? Let's close on this soon